test/saithrift/tests/sail3.py

```diff
@@ -1477,3 +1477,192 @@ def runTest(self):
+                    logging.debug("found expected packet from port %d" % destanation_ports[match_index])
+                    pkt_counter[match_index] += 1
+                    sport = random.randint(0,0xffff)
+                    dport = random.randint(0,0xffff)
+                        
+        #final uniform distribution check
+        logging.debug(pkt_counter)
+        logging.debug(first_rif_port)
+        lag_packets = sum(pkt_counter[1:first_rif_port])
+        lag_average = lag_packets/(len(self.lag_members) + 1)
+        logging.debug("the sum of packets through the lag is " + str(lag_packets))
+        logging.debug("the lag average for the lag is " + str(lag_average))
+        for stat_port in range(1,first_rif_port):
+            logging.debug( "PORT #"+str(stat_port)+":")
+            logging.debug(str(pkt_counter[stat_port]))
+            self.polarizationCheck(pkt_counter[stat_port],lag_average)
+        rifs_average = sum(pkt_counter)/(len(self.port_rifs) + 1)
+        logging.debug("lag average " + str(lag_average))
+        self.polarizationCheck(lag_packets,rifs_average)
+        for stat_port in range(first_rif_port,self.total_changing_ports):
+            logging.debug( "PORT #"+str(stat_port)+":")
+            logging.debug(str(pkt_counter[stat_port]))
+            self.polarizationCheck(pkt_counter[stat_port],rifs_average)
+    
+                 
+    def runTest(self):
+        """
+        For sai server, testing different lags with router
+        ---- Test for 16 ports minimun ----
+        Steps
+        1. Create virtual router, lag with rif and rif for src port
+        2. add ports to the lag and the rest of the ports connect to rifs
+        3. configure neighbors, nhops for all of the rifs
+        4. make ecmp route with all of the nhops
+        5. send packets from src port
+        6. check polarization check in the lag and in the ecmp
+        7. remove rifs, neighbors, nhops, lag members and route
+        8. repeat steps 3-7 with differnt numbers of lag members and rifs
+        8. clean up.
+        """
+       
+          
+        print
+        print "L3MultipleEcmpLagTest"
+        #general configuration 
+        random.seed(1)
+        switch_init(self.client)
+        self.src_port = port_list[0]
+
+        self.vr_id = sai_thrift_create_virtual_router(self.client, self.v4_enabled, self.v6_enabled)
+        rif_port_id = sai_thrift_create_router_interface(self.client, self.vr_id, 1, self.src_port, 0, self.v4_enabled, self.v6_enabled, '')
+        self.lag = self.client.sai_thrift_create_lag([])
+        sai_thrift_create_lag_member(self.client, self.lag, port_list[1])
+
+        self.lag_rif = sai_thrift_create_router_interface(self.client, self.vr_id, 1, self.lag, 0, self.v4_enabled, self.v6_enabled, '')
+        #create neighbor
+        sai_thrift_create_neighbor(self.client, self.addr_family, self.lag_rif, "10.10.16.1", self.mac_pool[15])
```

Comment:
making code modularized and readable is also important factor which makes the code more maintainable.